connect/src/com/telenor/connect/WellKnownAPI.java

```diff
@@ -28,5 +33,44 @@ public String getIssuer() {
         public Set<String> getNetworkAuthenticationTargetIps() {
             return networkAuthenticationTargetIps;
         }
+
+        protected WellKnownConfig(Parcel in) {
+            issuer = in.readString();
+            networkAuthenticationTargetIps = new HashSet<>();
+            int count = in.readInt();
+            for (int i = 0; i < count; i++) {
+                networkAuthenticationTargetIps.add(in.readString());
+            }
+        }
+
+        @Override
+        public int describeContents() {
+            return 0;
+        }
+
+        @Override
+        public void writeToParcel(Parcel dest, int flags) {
+            dest.writeString(issuer);
+            if (networkAuthenticationTargetIps == null) {
```

Comment:
So this doesn't make a distinction between a null networkAuthenticationTargetIps and an empty networkAuthenticationTargetIps so null is always deserialized to empty. Rest of the code seems to treat both those cases the same so it probably doesn't matter. If it's easy you can try making networkAuthenticationTargetIps always non-null so that case doesn't need to be handled.